request.go

```diff
@@ -602,14 +603,16 @@ func (n *DN) Equal(other *DN) bool {
 		n.JOILocality == other.JOILocality &&
 		n.JOIState == other.JOIState &&
 		n.JOICountry == other.JOICountry &&
-		n.BusinessCategory == other.BusinessCategory
+		n.BusinessCategory == other.BusinessCategory &&
+		n.SerialNumber == other.SerialNumber
 }
 
 // PKIXName converts a subject distinguished name into a pkix.Name object.
 func (n *DN) PKIXName() pkix.Name {
-	// Initialize name with common name.
+	// Initialize name with all single-value fields.
 	var name = pkix.Name{
-		CommonName: n.CommonName,
+		CommonName:   n.CommonName,
+		SerialNumber: n.SerialNumber,
```

Comment:
Why was this added here rather than adding it in the later section which "[copies] across fields with single values" (line 616/619)

Reply:
Yeah, that bit of code is a bit weird but its because both CommonName and SerialNumber are not arrarys in the pkix.Name struct, while the other fields are arrays.

Reply:
Ok, you probably need to update some comments then as this section claims to "Initialize [sic] all single-value fields." and then line 615/618 also claims to "Copy across fields with single values" which is partly where the confusion comes from